sdks/js-sdk/README.md

```diff
@@ -0,0 +1,150 @@
+# Abstract
+
+This document outlines what is needed to provide a set of the SDK's that can be used for JavaScript platforms, specifically Cordova, React Native & Web.
+
+## Goals
+
+### Primary Goals
+
+* Provide SDKs (Core, Metrics, Auth etc..) for Mobile Developers to allow them to integrate Cordova, React Native & Web applications into Mobile Services.
+
+### Secondary Goals
+
+* Reuse as much JavaScript as possible between the 3 different target platforms
+* Provide mechanisms for feature-parity with a similar API between all target platforms considering their differences
+
+## Problem Description
+
+The 3 target platforms (Cordova, React Native & Web) all support JavaScript as the development language. To develop Mobile Apps that integrate into Mobile Services, it is necessary to provide SDKs for Mobile Developers for the target platform & language. These SDKs need to be written & maintained. 
+
+## Proposed solution
+
+Provide JavaScript implementations (via a JavaScript SDK). This means the default way of implementing an SDK interface will be in JavaScript. However, this may not always be possible. 
+For the Cordova and React Native platforms we will provide a plugin that encapsulates the native code. This plugin conforms to an interface defined in the JS SDK that abstracts all the native platform specific code. Doing this will make Cordova, Web and React Native SDK's more consistent. We will bundle an
+implementation of that Interface for the Web Platform in the JS SDK. The plugins for Cordova and React Native will be separate repositories. The JS SDK requires the users to install the plugin for those Platforms and it will either try to auto-detect the platform, or require the user to pass in a `PlatformImpls` module when a service is instantiated.
+
+### Advantages
+
+- Ability to reuse code for Cordova, React Native & Web and work in a single repository
+- Can reuse JavaScript implementations for keycloak, push, sync.js etc.
+- No need to adjust native code (from Native SDKs) for plugin use. The native wrappers will not depend on the Android or iOS SDKs but provide their own implementations.
+
+## Architecture
```

Comment:
It could be nice to have a header per package (core, other sdks...) and explain that core would be required by the rest of packages.